metal/datasource_metal_reserved_ip_block.go

```diff
@@ -0,0 +1,169 @@
+package metal
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/packethost/packngo"
+)
+
+func dataSourceMetalReservedIPBlock() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceMetalReservedIPBlockRead,
+		Schema: map[string]*schema.Schema{
+			"block_id": {
```

Comment:
Perhaps we should just use `id` here. I've done this in the past.

I don't like the name `block_id` because it doesn't match anything else (the API field name, the resource name, the API resource endpoint name).

Reply:
This would also be consistent with:

https://github.com/equinix/terraform-provider-metal/pull/80/files#diff-1896ab87381448e70713530c6ffa8489b7da14b7b0615646fe86bff63d4677b5R50

Reply:
@displague I think that "id" is an implicit field of any TF resource or datasource. It's always there so you don't have to define it. It's set by SetId().

The block_id is a param for resource lookup. I can rename it if you want, just suggest a different name.

example DS with similar approach:
https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/data-sources/k8s_pool

Reply:
The only place where we take a datasource attribute that doesn't follow "full_resource_name_id" is spot_market_request where we call it "request_id".

The only place where we use "id" as the argument is ssh_key.

The [Terraform convention](https://www.terraform.io/docs/extend/best-practices/naming.html#naming) would be to match the field name in the API so TF documentation and the API documentation can follow closely.
> Most names in a Terraform provider will be drawn from the upstream API/SDK that the provider is using

I have a hard time leaning on _convention_ when it goes against _existing practice_.

Here are some examples where I used "id" as both optional and required parameters in another provider, which matched SetId after the resource was loaded:
* https://registry.terraform.io/providers/linode/linode/latest/docs/data-sources/image
* https://registry.terraform.io/providers/linode/linode/latest/docs/data-sources/domain_record

The field name for the underlying API response for `metal_reserved_ip_block` is `id`.

We may have inherited some confusion here from the API endpoint names (`/projects/{id}/ip`) not matching
g the API documented type name (`IPReservation`).   This datasource and the resource could have been called `metal_ip` if we were starting over. 

If we want to adopt full resource names across the board, we'll have to revisit the datasources for ssh_key and spot_market_request. 

If we want to consistently adopt `id` we'll have to revisit many more data sources.

This is a tough naming call, but I think we should go with `id` for this lookup argument. It's the obvious field name if you wanted to use this resource without documentation, or your were using the EM API docs as documentation.  `block_id` is not obvious because it does not match the resource name.

It seems other providers are inconsistent in naming, fwiw:
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ec2_transit_gateway#by-identifier
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/instance#argument-reference

Reply:
ok, I will change it to `id`